### ISSUE TITLE
Fix w3c validation warnings in viewer interface HTML

### ISSUE THREAD

**Issue by fedarko** [marbl/MetagenomeScope#79]
Looks like some of these are unavoidable due to Bootstrap elements/etc., but a number of these (for example, extraneous `</p>` tags, or the `min` property being used on inputs without an explicit numeric input type) are things that I can and should address.

**Comment by fedarko** [comment]
Huh. I did some more research and apparently it's considered invalid to nest `<div>` or `<p>` tags within `<p>` tags? It's a bit disconcerting that I didn't know that until just now.

Anyway, now that the cause of a lot of the validation errors is clear, I've made note of it in the HTML. will fix tomorrow

**Comment by fedarko** [comment]
(Likely unavoidable) errors -- see [here](http://getbootstrap.com/getting-started/#support-validators) for some context -- will update list as I continue to work on this --
- The validator registers an error that `autocomplete="off"` is present in `<input>` tags of type `radio` (for Bootstrap button groups that exhibit radio button behavior). The autocomplete attribute is needed to circumvent a bug in Firefox, so it looks like this error is unavoidable (unless we'd switch to using "normal" radio buttons, but since the current setup is working fine I don't think that'd be necessary).
- Similar errors as above, but with the `<input>` tags being inside `<button>` tags (switching to using `<label>` instead of `<button>` tags seems to exhibit other errors?)

**Comment by fedarko** [comment]
Remaining errors at this point:
- (6/14 errors) `Attribute autocomplete is only allowed when the input type is color, date, datetime-local, email, hidden, month, number, password, range, search, tel, text, time, url, or week.`
  - This error arises due to the use of `autocomplete="off"` on certain Bootstrap inputs (which is technically invalid). This is acceptable; see [this section](https://getbootstrap.com/docs/3.3/javascript/#buttons) of the Bootstrap docs for a description as to why this is necessary (circumventing a Firefox bug). For this reason, I'm not going to work on addressing these validation errors at this point.
- (another 6/14 errors) `The element input must not appear as a descendant of the button element.`
  - This error is due to me nesting `<input>` tags within `<button>` tags for the purposes of creating radio button groups that are styled like Bootstrap buttons (this is detailed [here](https://getbootstrap.com/docs/3.3/javascript/#buttons-checkbox-radio)).
  - Using `<label>` as the external tag (instead of `<button>` poses problems with both 1) disabling/enabling it and 2) W3C validation (but different errors come up -- complaining about how labels can't have `disabled` attributes, etc.)
  - Since this practice works fine -- and since the fancy radio buttons currently generated look a bit nicer than normal radio buttons -- I'm not going to work on addressing these validation errors at this point.
- (2/14 errors) `Attribute min not allowed on element input at this point.`
  - Declaring `min` (/`max`, eventually, although we do that through JS so HTML validation doesn't catch it) on the draw component input fields is technically invalid, since those input fields are of type `text` and not type `number`. I've moved discussion of this issue to #80, since addressing it (or deciding not to) will be somewhat involved.

That's it for the errors. There's also a warning I'm getting from the validator suggesting I avoid the `user-scalable=no` attribute of the `<meta name="viewport">` tag, but that serves an explicit purpose (and it's just a warning) so I'm not going to address that.

So, to sum things up: the only real thing left to consider addressing is the component selection input types, which will be taken care of in #80. Therefore, I'm closing this issue.
